### PR TITLE
DiabloExe::BaseItem, DiabloExe::Prefix, FAWorld::Inventory, draggable widgets and more

### PR DESCRIPTION
**Pull request by mloccy**

### CODE REVIEW

File: apps/freeablo/faworld/inventory.cpp [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+            case eqRING:
+                if(item.item.equipLoc == eqRING)
+                {
+                    if(!ringSlot) //ringSlot=0 => left hand ring
+                    {
+                        if(&this->leftRing == &emptyItem) return true;
+
+
+                    }
+                    else if(ringSlot) //ringSlot=0 => right hand ring
```
Comment (wheybags) [inline apps/freeablo/faworld/inventory.cpp]: should be ringSlot=1 in the comment?

File: resources/python/draggable.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+from docmanage import DocManager
+import inspect
+import pprint
+class DraggableWidget(object):
+    def __init__(self, document, xoffset, yoffset):
```
Comment (wheybags) [inline resources/python/draggable.py]: document isn't actually being used here

**Comment by mloccy** [comment]
This is now ready to be pulled I think

**Comment by mloccy** [comment]
There's still a bug in libRocket that means that textures wont show in the inventory, as far as I can tell librocket is not picking up background-image selectors when elements are dynamically created in python. Not sure if this would happen with non-cel images.

**Pull request merged**